crl/updater/updater.go

```diff
@@ -335,6 +346,21 @@ func (cu *crlUpdater) tickShard(ctx context.Context, atTime time.Time, issuerNam
 	cu.log.Infof(
 		"Generating CRL shard: id=[%s] numChunks=[%d]", crlID, len(chunks))
 
+	if features.Enabled(features.LeaseCRLShards) {
+		// Notify the database that we're working on this shard.
+		deadline, ok := ctx.Deadline()
+		if !ok {
+			return fmt.Errorf("context has no deadline")
+		}
+
+		_, err = cu.sa.LeaseCRLShard(ctx, &sapb.LeaseCRLShardRequest{
```

Comment:
`err` is being set, but there's no followup err check. The next error check that happens i on line 373 after `err` gets overridden. 

Should we have a log line in here to note that a specific shard has been selected so we can determine what branch it took through the SA - either a specific shard or the oldest shard?

Reply:
What's a little different about this method is that it uses a named return value (`err`) which Aaron is shadowing throughout and then checking on `defer`. See lines `336`-`344`.

Reply:
Maybe, to avoid confusion, we could choose a different name for this error?

Reply:
Huh, this should have been caught by a linter. Not checking this error is definitely a mistake.

Reply:
Ah interesting. I didn't catch that. I wrote this which helped me understand.
```
package main

import "fmt"

func incr(x int) (i int) {
	defer func() {
		i = x + 1
	}()
	return
}

func main() {
	fmt.Println(incr(2))
}
```

Reply:
@pgporada it appears you've caught a problem here, great work!